plugins/animate/animate.cpp

```diff
@@ -243,13 +292,78 @@ class wayfire_animation : public wf::singleton_plugin_t<animation_global_cleanup
         return {"none", 0};
     }
 
+    bool reversed(wayfire_view view, wf_animation_type type, std::string name,
+        int direction)
+    {
+        if (view->has_data(name))
+        {
+            auto data = view->get_data<animation_hook_base>(name);
+            if (data->get_direction() == direction)
+            {
+                data->reverse(type);
+                return true;
+            }
+        }
+
+        return false;
+    }
+
     template<class animation_t>
     void set_animation(wayfire_view view,
         wf_animation_type type, int duration)
     {
-        view->store_data(
-            std::make_unique<animation_hook<animation_t>>(view, duration, type),
-            animate_custom_data_id);
+        if (type == ANIMATION_TYPE_UNMAP)
+        {
+            view->take_ref();
+            view->take_snapshot();
+        }
+
+        if (type == ANIMATION_TYPE_MAP)
+        {
+            if (reversed(view, type, animate_custom_data_fire, 0) ||
```

Comment:
We should introduce constants for this, maybe

```
static constexpr int HIDDEN = 0;
...                  SHOWN  = 1;
```

Reply:
This seems like it belongs in an enum in wf-config but I can add some definitions for them in animate.

Reply:
Well the 0 does not necessarily indicate HIDDEN (in wf-config) so i'd say here's the place to put them.